nodejs/week3/homework/warmup/index.js

```diff
@@ -0,0 +1,56 @@
+const { json } = require("express");
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+app.get("/calculator/add", (req, res) => {
+    const requestedNum = Object.values(req.query).map(num => Number(num))
+    const addition = requestedNum.reduce((previousValue, currentValue) => previousValue + currentValue)
+    res.send({result: `${addition}`})
+})
+
+app.get("/calculator/subtraction", (req, res) => {
+    const requestedNum = Object.values(req.query).map(num => Number(num))
+    const subtraction = requestedNum.reduce((previousValue, currentValue) => previousValue - currentValue)
+    res.send({result: `${subtraction}`})
+})
+
+app.get("/calculator/multiply", (req, res) => {
+    const requestedNum = Object.values(req.query).map(num => Number(num))
+    const multiply = requestedNum.reduce((previousValue, currentValue) => previousValue * currentValue)
+    res.send({result: `${multiply}`})
+})
+
+app.get("/calculator/division", (req, res) => {
+    const requestedNum = Object.values(req.query).map(num => Number(num))
+    const division = requestedNum.reduce((previousValue, currentValue) => previousValue / currentValue)
+    res.send({result: `${division}`})
+})
+
+app.post("/calculator/add", (req, res) => {
+    console.log(req.query);
+    const requestedNum = Object.values(req.query).map(num => Number(num))
+    const addition = requestedNum.reduce((previousValue, currentValue) => previousValue + currentValue)
+    res.send({result: `${addition}`})
+})
+
+app.post("/calculator/subtraction", (req, res) => {
+    const requestedNum = Object.values(req.query).map(num => Number(num))
```

Comment:
This is a post request, but you're still accessing the query parameters, which are passed in as part of the url.
You need to read the parameters from the request body!
Here is some more information about the difference between GET and POST requests:
https://www.guru99.com/difference-get-post-http.html and https://www.postman.com/ is a useful tool to make POST requests and test your api.
Also, here is how you can access the body parameters in nodejs using express:
https://nodejs.dev/learn/get-http-request-body-data-using-nodejs

Reply:
Yes thank you I will watch the videos and correct it as soon as possible